src/main/scala/uncore/tilelink2/RegisterCrossing.scala

```diff
@@ -147,15 +147,18 @@ object AsyncRWSlaveRegField {
     slave_reset: Bool,
     width:  Int,
     init: Int,
+    name: Option[String] = None,
     master_allow: Bool = Bool(true),
     slave_allow: Bool = Bool(true)
   ): (UInt, RegField) = {
 
     val async_slave_reg = Module(new AsyncResetRegVec(width, init))
+    if (name.isDefined) {async_slave_reg.suggestName(name.get)}
```

Comment:
You're welcome to reject this as too-cute but I'd usually express something like the above as:
`name.foreach(async_slave_reg.suggestName(_))`

Basically you can treat the option as a (possibly empty) collection. See http://blog.tmorris.net/posts/scalaoption-cheat-sheet/ for more...